satellite/contact/endpoint.go

```diff
@@ -42,23 +49,23 @@ func (endpoint *Endpoint) CheckIn(ctx context.Context, req *pb.CheckInRequest) (
 
 	peerID, err := identity.PeerIdentityFromContext(ctx)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		return nil, rpcstatus.Error(rpcstatus.Unknown, errCheckInIdentity.New("failed to get ID from context: %v", err).Error())
 	}
 	nodeID := peerID.ID
 
 	err = endpoint.service.peerIDs.Set(ctx, nodeID, peerID)
 	if err != nil {
-		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+		return nil, rpcstatus.Error(rpcstatus.Unknown, errCheckInIdentity.New("failed to get ID from context: %v", err).Error())
```

Comment:
the message here is wrong, I think- it has actually failed to update the peer identity database. this is sufficiently different from the PeerIdentityFromContext case that it would be nice to have a different grpc code. I suggest maybe FailedPrecondition?

Reply:
oops. copy/paste fail. thanks for the recommendation, i will add FailedPrecondition status code and fix error message